src/js/osweb/items/loop.js

```diff
@@ -112,41 +138,46 @@ export default class Loop extends Item {
               this.matrix[cycle][name] = value
               break
             case 'fullfactorial':
-              this.matrix = fullfactorial(this.matrix)
-              // Set the number of cycles to the length of the generated matrix
-              this.vars.cycles = this.matrix.length
+              this._operations.push([fullfactorial, []])
+              // this.matrix = fullfactorial(this.matrix)
               break
             case 'shuffle':
-              this.matrix = shuffleVert(this.matrix, params)
+              this._operations.push([shuffleVert, [params]])
+              // this.matrix = shuffleVert(this.matrix, params)
               break
             case 'shuffle_horiz':
-              this.matrix = shuffleHoriz(this.matrix, params)
+              this._operations.push([shuffleHoriz, [params]])
+              // this.matrix = shuffleHoriz(this.matrix, params)
               break
             case 'slice':
-              this.matrix = this.matrix.slice(...params)
-              // Set the number of cycles to the length of the generated matrix
-              this.vars.cycles = this.matrix.length
+              this._operations.push([(matrix, args) => matrix.slice(...args), [params]])
+              // this.matrix = this.matrix.slice(...params)
               break
             case 'sort':
-              this.matrix = sortCol(this.matrix, ...params)
+              this._operations.push([sortCol, [...params]])
+              // this.matrix = sortCol(this.matrix, ...params)
               break
             case 'sortby':
-              this.matrix = sortBy(this.matrix, params)
+              this._operations.push([sortBy, [params]])
+              // this.matrix = sortBy(this.matrix, params)
               break
             case 'reverse':
-              this.matrix = reverseRows(this.matrix, params)
+              this._operations.push([reverseRows, [params]])
+              // this.matrix = reverseRows(this.matrix, params)
               break
             case 'roll':
-              this.matrix = roll(this.matrix, ...params)
+              this._operations.push([roll, [...params]])
+              // this.matrix = roll(this.matrix, ...params)
               break
             case 'weight':
-              this.matrix = weight(this.matrix, ...params)
-              // Set the number of cycles to the length of the generated matrix
-              this.vars.cycles = this.matrix.length
+              this._operations.push([weight, [...params]])
+              // this.matrix = weight(this.matrix, ...params)
               break
           }
         }
       }
+      // Set the number of cycles to the length of the generated matrix
+      this.vars.cycles = this.matrix.length

```

Comment:
Is this necessary here? I think this is set after the operations etc. have been applied and this can safely be removed here